### PR TITLE
Add label for SiPixelQualityRcd in RawToDigi

### PR DESCRIPTION
**Pull request by tsusa**
#### PR description:

This PR introduces a label for SiPixelQualityRcd used in RawToDigi ("forRawToDigi"). This allows to switch off different (number of) ROCs than are there in the unlabelled payload.
 
By default, variable UseQualityInfo, that governs SiPixelQuality usage, is set to false. A modifier is introduced to activate SiPixelQuality usage with the labelled payload (a new GT containing SiPixelQualityRcd with the label "forRawToDigi" is needed).

We plan to use this mechanism in the ReReco of Pilot Run data to suppress fake tracks arising from the noisy FPix ROCs.   

No changes are expected.

#### PR validation:

The PR has passed a basic set of tests (runTheMatrix.py -l limited -i all --ibeos).

### CODE REVIEW

**Comment by tsusa** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
Ok, so I understand correctly we need a new multi-IOV tag in the GT that has this label for the real PR testing. When the tag exists we can create the GT and then Tanja can include it in this PR

File: EventFilter/SiPixelRawToDigi/python/SiPixelRawToDigi_cfi.py [hunk 1]
```diff
@@ -14,6 +14,17 @@
     UsePhase1 = True
 )
 
+from Configuration.ProcessModifiers.siPixelQualityRawToDigi_cff import siPixelQualityRawToDigi
+siPixelQualityRawToDigi.toModify(siPixelDigis.cpu,  
+    UseQualityInfo = True,                         
+    SiPixelQualityLabel = 'forRawToDigi',
+)
+from CalibTracker.SiPixelESProducers.SiPixelQualityESProducer_cfi import siPixelQualityESProducer
+siPixelQualityRawToDigi.toModify(siPixelQualityESProducer,
```
Comment (slava77) [inline EventFilter/SiPixelRawToDigi/python/SiPixelRawToDigi_cfi.py:23]: modifiers have to be applied in the place of the module definitions.
Please remove here and apply in CalibTracker/SiPixelESProducers/python/SiPixelQualityESProducer_cfi.py

**Comment by slava77** [comment]
@tsusa 
please note the code review comment https://github.com/cms-sw/cmssw/pull/36255#pullrequestreview-817133616
(just trying to make sure that this gets to the pre3 deadline tomorrow on time).
Thank you.

**Comment by mmusich** [comment]
urgent

**Comment by mmusich** [comment]
the reason to mark this urgent is because it make a large difference when used in conjunction of the right payloads on the beam test ZeroBias data:

image
image
image

key:
   * black: reco as is
   * blue: this PR + masking a ROC level
   * orange: this PR + masking at module level

**Comment by tsusa** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
> @cmsbuild please test

@tsusa what's the point of this test if there is no tag in the GT that would be consumed by the changes in this PR?

**Comment by mmusich** [comment]
> what's the point of this test if there is no (/until there is) tag in the GT that would be consumed by the changes in this PR?

@tvami, by default this PR is not changing any physics output. If you want to test the new masking mechanism, you actually would also need to change the customization of he Pilot Beam RelVals introduced in https://github.com/cms-sw/cmssw/pull/36133 in order to use the process modifier `siPixelQualityRawToDigi` introduced here.
In addition to that, as discussed at the AlCa/DB meeting today some manipulation of several tags is needed. 
In the interest of moving forward fast towards the re-reco, would it be possible to consider the validation provided by @tsusa above (https://github.com/cms-sw/cmssw/pull/36255#issuecomment-981786399) [^1] and test the mechanism when a release with this feature is available via full-track AlCa validation?

[^1]:` ssh -NL 8080:localhost:8080 <user>@lxplus747.cern.ch`+ https://tinyurl.com/yxrvlknw

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/36255	02de84f 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass (the new variant of the code is not exercised)

**Comment by tvami** [comment]
> > what's the point of this test if there is no (/until there is) tag in the GT that would be consumed by the changes in this PR?
> 
> @tvami, by default this PR is not changing any physics output. If you want to test the new masking mechanism, you actually would also need to change the customization of the Pilot Beam RelVals introduced in #36133 in order to use the process modifier `siPixelQualityRawToDigi` introduced here. In addition to that, as discussed at the AlCa/DB meeting today some manipulation of several tags is needed. In the interest of moving forward fast towards the re-reco, would it be possible to consider the validation provided by @tsusa above ([#36255 (comment)](https://github.com/cms-sw/cmssw/pull/36255#issuecomment-981786399)) [1](#user-content-fn-1-149bad37fecd5e46a8363f7b14770901) and test the mechanism when a release with this feature is available via full-track AlCa validation?
> 

Ok sounds good, I'll sign now

**Comment by tvami** [comment]
+alca
 * see the further plan in https://github.com/cms-sw/cmssw/pull/36255#issuecomment-982015447

**Comment by tvami** [comment]
@perrotta @qliphy we are essentially fully signed

**Comment by qliphy** [comment]
+1

**Pull request merged**